www/pfSense-pkg-vHosts/files/usr/local/www/vhosts.php

```diff
@@ -0,0 +1,422 @@
+<?php
+/*******************************************************************************
+* vHosts package configuration page. Drops a list of the configured virtual
+* hosts or a form page for host configuration. When a form is posted back, the
+* host information is saved and the server config file is rebuilt.
+*
+* GET: No parameters:
+*   Drops list of configured vhosts with options to add/change/delete or toggle
+*   the disabled flag.
+*
+* GET: 'act' = 'add'
+*   Drops form for input of new host settings.
+*
+* GET: 'act' = 'chg', 'id' = n
+*   Drops form for update of host 'n' config.
+*   Form action returns 'id' value.
+*
+* GET: 'act' = 'del', 'id' = n
+*   Removes host 'n' config, update web config, and drops list page.
+*
+* GET: 'act' = 'dup', 'id' = n
+*   Loads the 'add' page with defaults loaded from host 'n' config.
+*   Form action returns 'srcid' set to 'id'.
+*
+* GET: 'act' = 'tog', 'id' = n
+*	Toggles Disabled flag for host 'n'.
+*
+* POST: 'id' = n (optional), 'srcid' = n (optional)
+*   Adds or updates host config, updates web config, and drops list page.
+*   If 'id' is empty but 'srcid' has a value, the new copy is inserted
+*   after the source host ID so it appears next in the list.
+* ------------------------------------------------------------------------------
+* Part of pfSense 2.3 and later (https://www.pfSense.org/).
+* Copyright (c) 2004-2016 Rubicon Communications, LLC (Netgate)
+*
+* Inspired by vhosts package originally written by Mark Crane.
+* Copyright (C) 2008 Mark J Crane
+* Copyright (C) 2015 ESF, LLC
+* Copyright (C) 2016 Softlife Consulting
+* All rights reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+require_once("guiconfig.inc");
+require_once("util.inc");
+require("/usr/local/pkg/vhosts.inc");
+
+$pgtitle = [ gettext("Services"), gettext("vHosts"), gettext("Hosts") ];
+$pgtabs  = 
+	[
+	[ gettext("Hosts"),        true,  "vhosts.php" ],
+	[ gettext("Certificates"), false, "vhosts_certs.php" ]
+	];
+
+/*******************************************************************************
+* Main Line
+*******************************************************************************/
+$a_vhosts = &$config['installedpackages']['vhosts']['config'];
+$act      = $_GET['act'];
+$id       = $_GET['id'];
```

Comment:
Make sure this `$id` value is sane (e.g. test against `is_numericint()`) or unset it here. The way it gets used later which could be a potential XSS vector.